File: src/leetCode/java/Solution409.java
package leetCode.java;

import java.util.*;

//O(n) O(Z), Z=26
class Solution409 {
    public int longestPalindrome(String s) {
        Set<Character> memo = new HashSet<>();
        int res = 0;
        for(char ch : s.toCharArray()){
            if(memo.contains(ch)){
                memo.remove(ch);
                res += 2;
            }else{
                memo.add(ch);
            }
        }
        return res + (memo.isEmpty() ? 0 : 1);
    }
}

File: src/leetCode/c++/Solution409.cpp
//O(n) O(Z), Z=26
class Solution409 {
public:
    int longestPalindrome(string s) {
        unordered_set<char> memo;
        int res = 0;
        for(char ch : s){
            if(memo.count(ch)){
                memo.erase(ch);
                res += 2;
            }else{
                memo.insert(ch);
            }
        }
        return memo.empty() ? res : res + 1;
    }
};